systemtest/src/main/java/io/strimzi/systemtest/parallel/TestSuiteNamespaceManager.java

```diff
@@ -0,0 +1,181 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.systemtest.parallel;
+
+import io.strimzi.systemtest.Constants;
+import io.strimzi.systemtest.Environment;
+import io.strimzi.systemtest.resources.kubernetes.NetworkPolicyResource;
+import io.strimzi.systemtest.utils.StUtils;
+import io.strimzi.test.k8s.KubeClusterResource;
+import io.strimzi.test.logs.CollectorElement;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.junit.jupiter.api.extension.ExtensionContext;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.function.Function;
+
+/**
+ * This class contains static information (additional namespaces will are needed for test suite) before the execution of @BeforeAll.
+ * By this we can easily prepare namespace in the AbstractST class and not in the the children.
+ */
+public class TestSuiteNamespaceManager {
+
+    private static final Logger LOGGER = LogManager.getLogger(TestSuiteNamespaceManager.class);
+    private static final boolean NAMESPACED_RBAC = Environment.isNamespaceRbacScope();
+    private static final Function<List<String>, List<String>> SUITE_NAMESPACES = (additionalSuiteNamespaces) -> {
+        if (NAMESPACED_RBAC) {
+            // rbac namespace we are gonna use INFRA_NAMESPACE (everything is in sequential execution)
+            return Collections.singletonList(Constants.INFRA_NAMESPACE);
+        } else {
+            return additionalSuiteNamespaces;
+        }
+    };
+
+    private static AtomicInteger counterOfNamespaces;
+    private static TestSuiteNamespaceManager instance;
+
+    /**
+     * Map @code{mapOfAdditionalNamespaces} which contains all additional namespaces for each test suite. It also
+     * uses @code{SUITE_NAMESPACES} function, which returns main namespace in case we have RBAC policy set to NAMESPACE
+     * instead of CLUSTER. Otherwise it returns list of additional namespaces.
+     */
+    private Map<String, List<String>> mapOfAdditionalNamespaces;
+
+    public synchronized static TestSuiteNamespaceManager getInstance() {
+        if (instance == null) {
+            instance = new TestSuiteNamespaceManager();
+            instance.constructMapOfAdditionalNamespaces();
+            counterOfNamespaces = new AtomicInteger(0);
+        }
+        return instance;
+    }
+
+    private void constructMapOfAdditionalNamespaces() {
+        this.mapOfAdditionalNamespaces = new HashMap<>();
```

Comment:
So when we add new package, we have to change this as well?

Reply:
Yes, we do.

Reply:
Do we have it documented somewhere?

Reply:
I have updated `testing.md`